tests/usbus/main.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (C) 2019 Koen Zandberg <koen@bergzand.net>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Tests USB interaction with USBUS
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <errno.h>
+
+#include "mutex.h"
+#include "periph/usbdev.h"
+#include "embUnit.h"
+
+#include "usbdev_mock.h"
+#include "usb/descriptor.h"
+#include "usb/usbus.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief Test case states
+ */
+typedef enum {
+    TESTPHASE_START,        /**< Initial start */
+    TESTPHASE_RESET,        /**< Signal USB RESET */
+    TESTPHASE_REQ_DEV_INIT, /**< Request the first part of the device descriptor */
+    TESTPHASE_RESET2,       /**< Second reset */
+    TESTPHASE_SET_ADDRESS,  /**< Configure Address */
+    TESTPHASE_REQ_DEV_FULL, /**< Request the full device descriptor */
+    TESTPHASE_FINAL,        /**< Testing done */
+} testphase_t;
+
+/**
+ * @brief Test control request phases
+ */
+typedef enum {
+    TEST_REQ_PHASE_IDLE,
+    TEST_REQ_PHASE_START_GET,
+    TEST_REQ_PHASE_START_SET,
+    TEST_REQ_PHASE_OUTDATA,
+    TEST_REQ_PHASE_INDATA,
+    TEST_REQ_PHASE_OUTACK,
+    TEST_REQ_PHASE_INACK,
+} test_req_phase_t;
+
+testphase_t phase = TESTPHASE_START;
+test_req_phase_t req_phase = TESTPHASE_START;
```

Comment:
```suggestion
test_req_phase_t req_phase = TEST_REQ_PHASE_IDLE;
```
?

Reply:
Oh wow… :flushed:

Reply:
Fixed

Reply:
Never mind, took a second attempt to actually fix it. Time to go to bed.